arch/lkl/kernel/signal.c

```diff
@@ -38,6 +38,17 @@ static void handle_signal(struct ksignal *ksig, struct ucontext *uctx)
     ksig->ka.sa.sa_handler(ksig->sig, (void*)&ksig->info, (void*)uctx);
 }
 
+/*
+    While you might think this should call move_signals_to_task, send_current_signals
+    it is different here in that only a specific signal is being requested.
+    This is the sync signal case and that signal will have just been forced into the
+    regular task queue of signals so the pre-existing code will work. See lkl_do_trap
+
+    Up for debate though is whether it would be better to either directly invoke
+    the handler in lkl_do_trap or here to have versions of move_signals_to_task, send_current_signals
+    that operate by scanning the list for signr, the given signal.
```

Comment:
I don't believe it would make sense to defer handling of sychronous signals, even if it is possible.

Reply:
As it stands it discards the other signals. Maybe I should capture the discarded ones to be sent later. Do not yet delivered signals usually survive a SEGV?

Reply:
It depends on whether the process survives a SEGV.  If it does, then it shouldn't lose a SIGIO just because it got a SIGSEGV.  There's no ordering between things like SIGIO and SIGSEGV.

Reply:
The trap path now preserves any other pending signals rather than discarding them.